src/Language/PureScript/Ide/Filter.hs

```diff
@@ -77,6 +79,15 @@ declarationFilter :: (IdeDeclaration -> Text -> Bool) -> Text -> [Module] -> [Mo
 declarationFilter predicate search =
   filterModuleDecls (flip predicate search)
 
+-- | Only keeps Identifiers in the given type declarations
+declarationTypeFilter :: [D.IdeDeclaration] -> Filter
+declarationTypeFilter [] = mkFilter identity
+declarationTypeFilter decls =
+    mkFilter $ filterModuleDecls filterDecls
+    where
+      filterDecls :: IdeDeclaration -> Bool
+      filterDecls decl = D.typeDeclarationForDeclaration decl `elem` decls
```

Comment:
@kRITZCREEK Moving `filterModuleDecls` into `Ide.Filter.Declaration` will cause a cycle dependency between `Ide.Filter` and `Ide.Filter.Declaration`. Because `Module` is defined in `Ide.Filter` and has to be imported into `Ide.Filter.Declaration` and `filterModuleDecls` has to be imported into `Ide.Filter`. To solve this cycle dependency we have to move less or more types etc. into other places (e.g. into a new module `Ide.Filter.Types` or similar), which "breaks" current module structure.

I would suggest that you move all these functions / types into other places as you like, but by creating another (new) issue for this. Just to avoid that current module structure will "destroyed" by my attempt (and not to waste to much time on this, of course). Thx!